### DISCUSSION TITLE
Possible reason for increased encoding time? (v1.4.0 --> v.1.5.2)

### DISCUSSION THREAD

**Discussion by LVCSRer** [ggml-org/whisper.cpp#1723] [Q&A] [answered]
Hello.

I was using v1.4.0 for developing iOS app, and recently found there was major update and applied to my app.

I use iPhone 12 pro and base.en model with Core ML.
I tested with my own recorded 15 seconds long audio, and below figure is average of 10 times iteration.

For Core ML without Metal setting, overall time was reduced due to decode time reduction, but encode time is increased. 

Does anyone know possible reason for this behavior? 

![image](https://github.com/ggerganov/whisper.cpp/assets/91233078/35fb206d-ca6b-464b-aaff-b8669fa6236b)

**Comment by ggerganov** [comment, marked as answer]
If you change this:

https://github.com/ggerganov/whisper.cpp/blob/ab0a8593c563ea78005a3c7a4d98b751e14ea16b/coreml/whisper-encoder.mm#L25-L29

To this:

```c
    // select which device to run the Core ML model on
    MLModelConfiguration *config = [[MLModelConfiguration alloc] init];
    //config.computeUnits = MLComputeUnitsCPUAndGPU;
    //config.computeUnits = MLComputeUnitsCPUAndNeuralEngine;
    config.computeUnits = MLComputeUnitsAll;
```

Does it help?

**Reply by LVCSRer** [reply]
Yes! Setting to both MLComputeUnitsCPUAndNeuralEngine and MLComputeUnitsAll reduces encode time by half for iPhone 12. 
Both setting require more time for first model initialization than MLComputeUnitsCPUAndGPU. (Maybe because of using ANE?) 

Thanks for the advice!
